code/Model/SiteTree.php

```diff
@@ -1633,6 +1633,11 @@ public function requireDefaultRecords()
         }
     }
 
+    private function hasURLSegment(): bool
+    {
+        return $this->URLSegment !== null && $this->URLSegment !== '';
```

Comment:
I guess excluding `false` is not necessary here, but would qualify as having a URL segment.

Reply:
I don't think it's necessary since it's unlikely to ever be `false` - but it doesn't hurt to add to avoid that edge case.
Added.